lib/libutils/ext/include/util.h

```diff
@@ -66,19 +67,27 @@
 }))
 
 /*
+ * ROUNDUP_DIV(x, y)
+ * ROUNDUP2_DIV(x, y)
+ *
  * Rounds up to the nearest multiple of y and then divides by y. Safe
  * against overflow, y has to be a power of 2.
  *
  * This macro is intended to be used to convert from "number of bytes" to
  * "number of pages" or similar units. Example:
  * num_pages = ROUNDUP_DIV(num_bytes, SMALL_PAGE_SIZE);
+ *
  */
-#define ROUNDUP_DIV(x, y) (__extension__({ \
-	typeof(x) __roundup_x = (x); \
-	typeof(y) __roundup_mask = (typeof(x))(y) - 1; \
-	\
-	(__roundup_x / (y)) + (__roundup_x & __roundup_mask ? 1 : 0); \
-}))
+#define ROUNDUP_DIV(x, y) \
+	(__extension__({ \
+		typeof(x) __roundup_x = (x); \
+		typeof(y) __roundup_mask = (typeof(x))(y) - 1; \
+		\
+		assert(IS_POWER_OF_TWO(y)); \
```

Comment:
This is inconsistent with the other `ROUNDUP*()` macros, those without a `2` should work with any alignment or granule.

Reply:
Agree. I'll make a dedicated relaxed `ROUNDUP_DIV()` macro.